### ISSUE TITLE
Watering the twisted.web2 docstring desert

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1919]
|<img alt="edsuom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| edsuom reported|
|-|-|
|Trac ID|trac#1919|
|Type|defect|
|Created|2006-07-12 21:12:35Z|

twisted.web2.client.http is devoid of docstrings. There are probably other equally important twisted.web2 modules that need docstring TLC, too.

<details><summary>Searchable metadata</summary>

```
trac-id__1919 1919
type__defect defect
reporter__edsuom edsuom
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1152738755000000 1152738755000000
changetime__1153009921000000 1153009921000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid commented|
|-|-|

I've written docstrings for web2 client, vhost, some things in http and some things in http_headers.  The focus is on user APIs.

More doc strings could be written but I'd like to go ahead and merge and close this bug so it doesn't get stagnant.

branched to [branches/web2-docstring-1919](branches/web2-docstring-1919)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen set owner to @dreid|
|-|-|

These all read fine.  Fixed a few typos, that's about it.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen commented|
|-|-|

Crap, should have actually run the tests.  "no module named interfaces" ... I will fix, don't check in yet.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen commented|
|-|-|

OK, tests pass in r17451.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set status to `closed`|
|-|-|

(In [17544]) Docstrings for various userfacing methods and classes.

This also contains a few extra lines for readability sake.

Reviewer: MFen
Author: dreid
Fixes [#1919](https://github.com/twisted/twisted/issues/1919)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
